src/App.tsx

```diff
@@ -15,36 +17,100 @@
   'Garlic',
 ];
 
+let goods = [...goodsFromServer];
+let isReverseCLicked = false;
+
 export const App: React.FC = () => {
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortGoods, setSortGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function setSort() {
+    if (sortGoods === ALPHABETICALLY) {
+      goods = [...goodsFromServer].sort((goodA, goodB) =>
+        goodA.localeCompare(goodB),
+      );
+    }
+
+    if (sortGoods === BY_LENGTH) {
+      goods = [...goodsFromServer].sort(
+        (goodA, goodB) => goodA.length - goodB.length,
+      );
+    }
+
+    if (reverse) {
+      goods.reverse();
+    }
+  }
+
+  if (sortGoods === RESET) {
+    goods = [...goodsFromServer];
+  }
```

Comment:
[CODE QUALITY] This code is executed on each render, but should only be executed when `sortGoods` is `RESET`. Consider moving this inside a `useEffect` hook with `sortGoods` as a dependency.